pkg/probe/http/http.go

```diff
@@ -28,15 +29,18 @@ import (
 )
 
 func New() HTTPProber {
-	return HTTPProber{&http.Client{}}
+	client := &http.Client{}
+	return HTTPProber{client, func(timeout time.Duration) { client.Timeout = timeout }}
 }
 
 type HTTPProber struct {
-	client HTTPGetInterface
+	client     HTTPGetInterface
+	setTimeout func(timeout time.Duration)
 }
 
 // Probe returns a ProbeRunner capable of running an http check.
-func (pr *HTTPProber) Probe(host string, port int, path string) (probe.Status, error) {
+func (pr *HTTPProber) Probe(host string, port int, path string, timeout time.Duration) (probe.Status, error) {
+	pr.setTimeout(timeout)
```

Comment:
Hmm, I was thinking that req.Close() was something but it's not. You'd need to close the underlying tcp connection which should cause client.Do() to return with an error. Rereading the doc, http.Transport.CancelRequest() should work but to get one form the an http.Client requires a type assertion. Instead of `defer req.Close()` maybe:
```
if t, ok := client.Transport.(*http.Transport); ok {
	defer t.CancelRequest(req)
}
```
which is also not pretty and requires some interface changing HTTPGetInterface but shouldn't leak tcp connections or goroutines. At that point you don't even need an http.Client though, just an http.Transport. That's just one option though, there may be a better.